### PR TITLE
Improve instructions for the release process

### PR DESCRIPTION
**Pull request by xinrong-meng**
Improve instructions for the release process, specifically, the `Finalize the release` step.

That includes
- point out which steps have been automated
- details to pay attention to
- typo fixing
- more clear steps by naming

### CODE REVIEW

**Comment by xinrong-meng** [comment]
Would you please review it? @dongjoon-hyun @HyukjinKwon @gengliangwang @viirya  @ueshin @gatorsmile

**Review by dongjoon-hyun — CHANGES_REQUESTED** [review]
CRAN description is correct?

**Review by dongjoon-hyun — APPROVED** [review]
CRAN description is correct?

**Comment by xinrong-meng** [comment]
We should submit the form linked in the description. More instructions are given when filling out the form.
I was able to fill it out independently without "further manual steps from PMC".
Would you prefer to remove the last sentence? @dongjoon-hyun

File: release-process.md [hunk 1]
```diff
@@ -186,6 +186,18 @@ that looks something like `[VOTE][RESULT] ...`.
 
 <h3>Finalize the release</h3>
 
+Note that `dev/create-release/do-release-docker.sh` script (`finalize` step ) automates most of the following steps **except** for:
+- Publish to CRAN
```
Comment (viirya) [inline release-process.md:190]: PyPI?

File: release-process.md [hunk 2]
```diff
@@ -186,6 +186,18 @@ that looks something like `[VOTE][RESULT] ...`.
...
+- Update the configuration of Algolia Crawler
+- Remove old releases from Mirror Network
+- Update the rest of the Spark website
+- Create and upload Spark Docker Images
+- Create an announcement
+
+Please manually verify the result after each step.
+
+<h4>Upload to Apache release directory</h4>
```
Comment (viirya) [inline release-process.md:199]: Isn't this step also manual?

**Review by viirya — APPROVED** [review]

**Comment by srowen** [comment]
Merged to asf-site

File: release-process.md (hunk 1 above)
Comment (xinrong-meng) [reply release-process.md:190]: "Uploading PySpark to PyPi" is automated.

File: release-process.md (hunk 2 above)
Comment (xinrong-meng) [reply release-process.md:199]: "Moving Spark binaries to the release directory" is automated.

**Comment by xinrong-meng** [comment]
Thanks @srowen !

File: release-process.md (hunk 2 above)
Comment (xinrong-meng) [reply release-process.md:199]: See more at https://github.com/xinrong-meng/spark/blob/master/dev/create-release/release-build.sh#L97. : )

File: release-process.md (hunk 1 above)
Comment (viirya) [reply release-process.md:190]: Hm? Last time I did release I remember I still need to do `twine upload`, has it changed?

EDIT: it was changed by SPARK-35872.

File: release-process.md (hunk 2 above)
Comment (viirya) [reply release-process.md:199]: Oh, it was changed by SPARK-35872. 👍

**Pull request closed without merging**